azurerm/resource_arm_servicebus_topic.go

```diff
@@ -107,11 +119,29 @@ func resourceArmServiceBusTopicCreate(d *schema.ResourceData, meta interface{})
 	namespaceName := d.Get("namespace_name").(string)
 	location := d.Get("location").(string)
 	resGroup := d.Get("resource_group_name").(string)
+	status := d.Get("status").(string)
+
+	enableBatchedOps := d.Get("enable_batched_operations").(bool)
+	enableExpress := d.Get("enable_express").(bool)
+	enableFiltering := d.Get("enable_filtering_messages_before_publishing").(bool)
+	enablePartitioning := d.Get("enable_partitioning").(bool)
+	maxSize := int64(d.Get("max_size_in_megabytes").(int))
+	requiresDuplicateDetection := d.Get("requires_duplicate_detection").(bool)
+	supportOrdering := d.Get("support_ordering").(bool)
 
 	parameters := servicebus.TopicCreateOrUpdateParameters{
-		Name:            &name,
-		Location:        &location,
-		TopicProperties: &servicebus.TopicProperties{},
+		Name:     &name,
+		Location: &location,
+		TopicProperties: &servicebus.TopicProperties{
+			Status:                            servicebus.EntityStatus(status),
+			EnableBatchedOperations:           &enableBatchedOps,
+			EnableExpress:                     &enableExpress,
+			FilteringMessagesBeforePublishing: &enableFiltering,
+			EnablePartitioning:                &enablePartitioning,
+			MaxSizeInMegabytes:                &maxSize,
+			RequiresDuplicateDetection:        &requiresDuplicateDetection,
+			SupportOrdering:                   &supportOrdering,
```

Comment:
I think we'd benefit from having a provider-wide helper for creating these pointers here. AWS has its own in the SDK (`aws.String()`, `aws.Bool()` etc.) and I built similar ones in K8S: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/master/kubernetes/structures.go#L124-L142

It may look like a nitpick, but you can effectively reduce the number of lines to half here.